### PR TITLE
Flow + Metadata + Query wrapper library

### PR DESCRIPTION
**Pull request by tlrobinson**

### CODE REVIEW

File: frontend/src/metabase/meta/memoize.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+
+let memoized = new WeakMap();
```
Comment (salsakran) [inline frontend/src/metabase/meta/memoize.js]: Is there a TTL here? How often will this get refreshed?

Comment (tlrobinson) [inline frontend/src/metabase/meta/memoize.js]: They will be GC'd when the parent metadata instance is GC'd thanks to the way [WeakMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap) works.

Putting a `memoized` map on each of the root metadata instances would have the same effect.

Comment (salsakran) [inline frontend/src/metabase/meta/memoize.js]: I'm thinking more in terms of when do changes in the metadata get reflected. Would I need to refresh the entire page to kill off the parent instance?

**Comment by salsakran** [comment]
is it possible to turn the demo.js files into unit tests or is that a pain?

**Comment by tlrobinson** [comment]
Yup, in progress.

**Comment by tlrobinson** [comment]
Ok, ready to go.

**Pull request merged**